discovery_child_development/notebooks/04_train_initial_model.py

```diff
@@ -0,0 +1,444 @@
+# * Log the model and key metrics on wandb
+
+# %%
+# Load libraries
+import pandas as pd
+import numpy as np
+import os
+import pickle
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import (
+    accuracy_score,
+    precision_score,
+    recall_score,
+    f1_score,
+    hamming_loss,
+    jaccard_score,
+    classification_report,
+)
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+## plotting libraries
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import wandb
+
+## nesta ds
+from nesta_ds_utils.loading_saving import S3
+
+## project code
+from discovery_child_development import PROJECT_DIR, logging
+from discovery_child_development.utils import bigquery, labelling_utils
+from discovery_child_development.utils import wandb as wb
+from discovery_child_development.utils.io import import_config
+
+bigquery.find_credentials("GOOGLE_SHEETS_CREDENTIALS")
+
+## constants
+
+S3_BUCKET = os.environ["S3_BUCKET"]
+PARAMS = import_config("config.yaml")
+CONCEPT_IDS = "|".join(PARAMS["openalex_concepts"])
+INPUT_PATH = "data/openAlex/processed/"
+DATA_PATH_LOCAL = os.path.join(PROJECT_DIR, "inputs", "data")
+FIG_PATH = os.path.join(PROJECT_DIR, "outputs", "figures")
+MODEL_PATH = os.path.join(PROJECT_DIR, "outputs", "models")
+SEED = 42
+
+## variables
+SCORE_THRESHOLD = 0.3  # we will remove any concepts (and corresponding subcategories) assigned with less than 0.3 confidence by the OpenAlex algorithm
+
+
+# %%
+# helpful functions
+class MostCommonClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(self, labels):
+        self.labels = labels
+
+    def fit(self, X, y=None):
+        # Nothing happens here because it doesn't learn from the data
+        return self
+
+    def predict(self, X):
+        # Returns the most common label combination for all input
+        return [self.labels for _ in range(len(X))]
+
+
+def create_average_metrics(Y_test, Y_pred, average="samples"):
+    # Accuracy
+    accuracy = accuracy_score(Y_test, Y_pred)
+
+    # Precision
+    precision = precision_score(
+        Y_test, Y_pred, average=average
+    )  # Use 'samples' for multi-label
+
+    # Recall
+    recall = recall_score(
+        Y_test, Y_pred, average=average
+    )  # Use 'samples' for multi-label
+
+    # F1-Score
+    f1 = f1_score(Y_test, Y_pred, average=average)  # Use 'samples' for multi-label
+
+    # Hamming Loss
+    hamming = hamming_loss(Y_test, Y_pred)
+
+    # Jaccard Score
+    jaccard = jaccard_score(
+        Y_test, Y_pred, average=average
+    )  # Use 'samples' for multi-label
+
+    results = {
+        "accuracy": accuracy,
+        "precision": precision,
+        "recall": recall,
+        "f1": f1,
+        "hamming": hamming,
+        "jaccard": jaccard,
+    }
+
+    return results
+
+
+def find_most_frequent_labels(df, label_col="sub_category", head=20):
```

Comment:
```suggestion
def find_most_frequent_labels(df: pd.DataFrame, label_col: str="sub_category", head: int=20) -> Tuple[pd.DataFrame, str]:
```

(Actually don't remember from top of my head if this is the right way to describe Tuple outputs)